tsconfig.base.json

```diff
@@ -9,16 +9,15 @@
     "noEmit": true,
     "declarationMap": true,
     "esModuleInterop": true,
-    "module": "commonjs",
-    "moduleResolution": "node",
+    "module": "Node18",
+    "moduleResolution": "node16",
```

Comment:
commonjs & node no longer accurately model nodejs module resolution rules, so they're effectively deprecated.
They would've highlighted (though perhaps not errored on) the primary issue patched here.

Reply:
This also stays @Node18 despite keeping support for ESLint 8.